django/db/migrations/operations/models.py

```diff
@@ -929,6 +929,15 @@ def migration_name_fragment(self):
     def reduce(self, operation, app_label):
         if isinstance(operation, RemoveIndex) and self.index.name == operation.name:
             return []
+        if isinstance(operation, RenameIndex) and self.index.name == operation.old_name:
+            return [
+                AddIndex(
+                    model_name=self.model_name,
+                    index=models.Index(
+                        name=operation.new_name, fields=self.index.fields
+                    ),
```

Comment:
We cannot create a new `Index` here as it's more complicated. `Index` can have `expressions` instead of `fields`, be a custom subclass, etc.

Reply:
Fixed.